### PR TITLE
feat(cargo-shuttle): beta updated local run

### PR DESCRIPTION
**Pull request by jonaro00**
extends https://github.com/shuttle-hq/shuttle/pull/1844

tested local run with:
rocket/hello-world
rocket/postgres
rocket/secrets

### CODE REVIEW

File: cargo-shuttle/src/lib.rs [hunk 1]
```diff
@@ -1416,9 +1435,7 @@ impl Shuttle {
                 service_name.as_str(),
                 false,
                 false,
-                // Set beta to false to avoid breaking local run with beta changes.
-                // TODO: make local run compatible with --beta.
-                false
```
Comment (oddgrd) [inline cargo-shuttle/src/lib.rs:1421]: Is the comment no longer accurate?

File: cargo-shuttle/src/lib.rs [hunk 2]
```diff
@@ -1787,6 +1820,182 @@ impl Shuttle {
...
+        );
+        let mut runtime = tokio::process::Command::new(
+            dunce::canonicalize(runtime_executable).context("canonicalize path of executable")?,
+        )
+        .current_dir(&service.workspace_path)
+        .args(["--run"])
+        .envs([
+            ("SHUTTLE_BETA", "true"),
+            ("SHUTTLE_PROJECT_ID", "proj_LOCAL"),
+            ("SHUTTLE_PROJECT_NAME", "TODO"),
```
Comment (oddgrd) [inline cargo-shuttle/src/lib.rs:1888]: What is the plan for this, and does the user see it?

File: cargo-shuttle/src/lib.rs [hunk 3]
```diff
@@ -1787,6 +1820,182 @@ impl Shuttle {
...
+                }))
+            }
+        });
+        let server = Server::bind(&api_addr).serve(make_svc);
+        tokio::spawn(async move {
+            if let Err(e) = server.await {
+                eprintln!("Provisioner server error: {}", e);
+                exit(1);
+            }
+        });
```
Comment (oddgrd) [inline cargo-shuttle/src/lib.rs:1866]: Should we abstract this into a BetaProvisionerServer struct?

**Review by oddgrd — APPROVED** [review]

File: cargo-shuttle/src/lib.rs (hunk 1 above)
Comment (jonaro00) [reply cargo-shuttle/src/lib.rs:1421]: Updating resource tables (if we want tables at all) for beta is still TODO, so whatever happens here will change.

File: cargo-shuttle/src/lib.rs (hunk 2 above)
Comment (jonaro00) [reply cargo-shuttle/src/lib.rs:1888]: TY for reminder. It's passed into ResourceFactory and Metadata.

**Pull request merged**